src/encoding.rs

```diff
@@ -552,6 +564,24 @@ impl EncodeCounterValue for f64 {
     }
 }
 
+impl EncodeCounterValue for u32 {
+    fn encode(&self, encoder: &mut CounterValueEncoder) -> Result<(), std::fmt::Error> {
+        encoder.encode_u64(*self as u64)
+    }
+}
+
+impl EncodeCounterValue for i32 {
```

Comment:
A counter can never be negative. Why implement `EncodeCounterValue` for `i32`?

Reply:
Fixed.